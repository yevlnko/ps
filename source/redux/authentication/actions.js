// Instruments
import { types } from "./types";

export const authActions = Object.freeze({
    login: (credentials) => ({
        type:    types.LOGIN,
        payload: credentials,
    }),
    loginSuccess: () => ({
        type: types.LOGIN_SUCCESS,
    }),
    loginFail: (error) => ({
        type:    types.LOGIN_FAIL,
        payload: error,
        error:   true,
    }),

    authenticate: (token) => ({
        type:    types.AUTHENTICATE,
        payload: token,
    }),
    authenticateSuccess: () => ({
        type: types.AUTHENTICATE_SUCCESS,
    }),
    authenticateFail: (error) => ({
        type:    types.AUTHENTICATE_FAIL,
        payload: error,
        error:   true,
    }),

    signup: (credentials) => ({
        type:    types.SIGNUP,
        payload: credentials,
    }),
    signupSuccess: () => ({
        type: types.SIGNUP_SUCCESS,
    }),
    signupFail: (error) => ({
        type:    types.SIGNUP_FAIL,
        payload: error,
        error:   true,
    }),

    logout: () => ({
        type: types.LOGOUT,
    }),
    logoutSuccess: () => ({
        type: types.LOGOUT_SUCCESS,
    }),
    logoutFail: (error) => ({
        type:    types.LOGOUT_FAIL,
        payload: error,
        error:   true,
    }),
});
