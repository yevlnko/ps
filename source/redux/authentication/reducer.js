// Core
import { Map } from 'immutable';

// Instruments
import { types } from "./types";

const initialState = Map({
    isAuthenticated: false,
});

export const authReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.LOGIN_SUCCESS:
        case types.SIGNUP_SUCCESS:
        case types.AUTHENTICATE_SUCCESS:
            return state.set('isAuthenticated', true);

        case types.LOGOUT_SUCCESS:
            return state.set('isAuthenticated', false);


        default:
            return state;
    }
};
