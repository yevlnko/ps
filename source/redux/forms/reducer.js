// Core
import { combineForms } from 'react-redux-form';
import { token } from "../../config/index";
import { dev } from "../../init/middleware/index";

export const formsReducer = combineForms({
    signup: {
        firstName: 'Yevhenii',
        lastName:  'Liashenko',
        email:     'test@test.com',
        password:  '123123',
        invite:    dev ? token : '',
    },
    login: {
        email:    '',
        password: '',
    },
    user: {
        profile: {
            firstName: '',
            lastName:  '',
            avatar:    [],
        },
        password: {
            oldPassword: '',
            newPassword: '',
        },
    },
}, 'forms');
