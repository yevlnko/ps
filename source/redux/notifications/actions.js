// Instruments
import { types } from "./types";
import { v4 } from "uuid";

export const notificationsActions = Object.freeze({
    showNotification: (error) => ({
        type:    types.SHOW_NOTIFICATION,
        payload: {
            id: v4(),
            error,
        },
    }),
    hideNotification: (id) => ({
        type:    types.HIDE_NOTIFICATION,
        payload: id,
    }),
});
