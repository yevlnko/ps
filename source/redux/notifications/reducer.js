// Core
import { List } from "immutable";

// Instruments
import { types } from "./types";

const initialState = List();

export const notificationsReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.SHOW_NOTIFICATION:
            return state.size < 2 ? state.push(action.payload) : state;

        case types.HIDE_NOTIFICATION:
            return state.filter((notification) => {
                return notification.id !== action.payload;
            });

        default:
            return state;
    }
};
