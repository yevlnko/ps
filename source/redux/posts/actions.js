// Instruments
import { types } from "./types";

export const postsActions = Object.freeze({
    fetchPosts: () => ({
        type: types.FETCH_POSTS,
    }),
    fetchPostsFail: (error) => ({
        type:    types.FETCH_POSTS_FAIL,
        payload: error,
        error:   true,
    }),
    fetchPostsSuccess: (posts) => ({
        type:    types.FETCH_POSTS_SUCCESS,
        payload: posts,
    }),
    createPost: (post) => ({
        type:    types.CREATE_POSTS,
        payload: post,
    }),
    removePost: (id) => ({
        type:    types.REMOVE_POSTS,
        payload: id,
    }),
    likePost: (data) => ({
        type:    types.LIKE_POST,
        payload: data,
    }),
    unlikePost: (data) => ({
        type:    types.UNLIKE_POST,
        payload: data,
    }),
});
