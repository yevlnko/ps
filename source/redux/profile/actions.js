// Instruments
import { types } from "./types";

export const profileActions = Object.freeze({
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
    clearProfile: () => ({
        type: types.CLEAR_PROFILE,
    }),
    updateAvatar: (avatar) => ({
        type:    types.UPDATE_AVATAR,
        payload: avatar,
    }),
});
