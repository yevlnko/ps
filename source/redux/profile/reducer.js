// Core
import { Map } from 'immutable';

// Instruments
import { types } from "./types";

const initialState = Map({
    id:        '',
    avatar:    '',
    firstName: '',
    lastName:  '',
});

export const profileReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.FILL_PROFILE:
            return state.merge(action.payload);

        case types.CLEAR_PROFILE:
            return initialState;

        case types.UPDATE_AVATAR:
            return state.set('avatar', action.payload);


        default:
            return state;
    }
};
