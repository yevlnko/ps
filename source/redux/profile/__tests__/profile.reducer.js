// Instruments
import { profileReducer } from "../reducer";
import { profileActions } from "../actions";

describe("profile Reducer:", () => {
    test("should return initial State", () => {
        expect(profileReducer(undefined, { type: "@@INIT" })).toMatchSnapshot();
    });

    test("should handle FILL_PROFILE", () => {
        expect(
            profileReducer(
                undefined,
                profileActions.fillProfile(__.userProfile)
            )
        ).toMatchSnapshot();
    });

    test("should handle CLEAR_PROFILE", () => {
        expect(
            profileReducer(undefined, profileActions.clearProfile())
        ).toMatchSnapshot();
    });

    test("should handle UPDATE_AVATAR", () => {
        expect(
            profileReducer(undefined, profileActions.updateAvatar([]))
        ).toMatchSnapshot();
    });
});
