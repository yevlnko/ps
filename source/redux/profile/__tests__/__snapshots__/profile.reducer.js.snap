// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile Reducer: should handle CLEAR_PROFILE 1`] = `
Immutable.Map {
  "id": "",
  "avatar": "",
  "firstName": "",
  "lastName": "",
}
`;

exports[`profile Reducer: should handle FILL_PROFILE 1`] = `
Immutable.Map {
  "id": "TEST_ID",
  "avatar": "TEST_AVATAR",
  "firstName": "Walter",
  "lastName": "White",
  "token": "TEST_TOKEN",
}
`;

exports[`profile Reducer: should handle UPDATE_AVATAR 1`] = `
Immutable.Map {
  "id": "",
  "avatar": Array [],
  "firstName": "",
  "lastName": "",
}
`;

exports[`profile Reducer: should return initial State 1`] = `
Immutable.Map {
  "id": "",
  "avatar": "",
  "firstName": "",
  "lastName": "",
}
`;
